app/src/main/kotlin/compiler/intermediate_form/ControlFlow.kt

```diff
@@ -43,6 +45,14 @@ object ControlFlow {
             )
         }
 
+        val variableAccessGenerators: ReferenceMap<VariablesOwner, VariableAccessGenerator> = run {
+            val result = ReferenceHashMap<VariablesOwner, VariableAccessGenerator>()
+
+            result.putAll(functionDetailsGenerators)
+            result[VariablePropertiesAnalyzer.GlobalContext] = GlobalVariablesAccessGenerator(variableProperties)
+            result
+        }
+
```

Comment:
Creating `GlobalVariablesAccessGenerator` object multiple times is not a good idea. Aside from performance concerns, according to Java docs (I couldn't find such information for Kotlin), HashMap "... makes no guarantees as to the order of the map...", which means we could end up with inconsistent offsets in different expressions. I suggest making that object of type `GlobalVariablesAccessGenerator` a parameter of `createGraphForExpression`.